### PR TITLE
Cancel reservations - ASU-1356

### PR DESCRIPTION
**Pull request by juhakujala**

### CODE REVIEW

File: src/components/reservations/ReservationCancelModal.tsx [hunk 1]
```diff
@@ -37,14 +43,28 @@ const ReservationCancelModal = (): JSX.Element | null => {
...
+  const handleFormCallback = async (formData: ReservationCancelFormData) => {
+    if (!postReservationCancelLoading) {
+      setIsLoading(true);
+
+      try {
+        // Send reservation cancel form data to API
+        await cancelApartmentReservation({ formData, reservationId: reservation.id, projectId: projectId })
+          .unwrap()
+          .then(() => {
+            toast.show({ type: 'success', content: t(`${T_PATH}.cancelledSuccessfully`) });
```
Comment (tuomas777) [inline src/components/reservations/ReservationCancelModal.tsx:55]: I think this is missing from translations?

File: src/i18n/en.json [hunk 2]
```diff
@@ -334,42 +334,52 @@
...
+      "LATE_PAYMENT_INTEREST": "Late payment interest",
+      "PAYMENT_1": "Payment 1",
+      "PAYMENT_2": "Payment 2",
+      "PAYMENT_3": "Payment 3",
+      "PAYMENT_4": "Payment 4",
+      "PAYMENT_5": "Payment 5",
+      "PAYMENT_6": "Payment 6",
+      "PAYMENT_7": "Payment 7",
+      "REFUND": "Refund",
+      "RESERVATION_FEE": "Feservation fee",
```
Comment (tuomas777) [inline src/i18n/en.json:350]:
```suggestion
      "RESERVATION_FEE": "Reservation fee",
```

**Review by tuomas777 — APPROVED** [review]
Added a couple of minor comments.

In general very great cancelling! 👍 🦑

File: src/i18n/en.json (hunk 2 above)
Comment (juhakujala) [reply src/i18n/en.json:350]: 😂 👍

File: src/components/reservations/ReservationCancelModal.tsx (hunk 1 above)
Comment (juhakujala) [reply src/components/reservations/ReservationCancelModal.tsx:55]: Yeah, I noticed that after doing the reservation creation feature and it is included in that PR #57 already.

File: src/enums.ts [hunk 3]
```diff
@@ -54,10 +54,9 @@ export enum ApartmentReservationStates {
   SUBMITTED = 'submitted',
 }
 
-// TODO: Re-define these after API is ready
 export enum ReservationCancelReasons {
-  RESERVATION_AGREEMENT_CANCELED = 'reservation_agreement_canceled',
-  RESERVATION_CANCELED = 'reservation_canceled',
+  CANCELED = 'canceled',
+  CONTRACT_TERMINATED = 'contract_terminated',
```
[the remark this answers is not in the document]
Comment (juhakujala) [reply src/enums.ts:59]: I think that `reservation_agreement_canceled` was just a term that I came up with when doing the dummy version of the component. We can use that if you want to 👍

File: src/components/reservations/ReservationCancelModal.tsx (hunk 1 above)
Comment (juhakujala) [reply src/components/reservations/ReservationCancelModal.tsx:55]: Added the translations

File: src/enums.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (juhakujala) [reply src/enums.ts]: Kept the original enum name

**Pull request merged**